lib/HttpSignatureError.js

```diff
@@ -0,0 +1,12 @@
+/*!
+ * Copyright (c) 2018 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+module.exports = class HTTPSignatureError extends Error {
```

Comment:
Will it cause any confusion having the same class name as the error in `http-signature-header`.  Doesn't look like the class is exposed in either API.

Reply:
Isn't this the same as BedrockError which is used across multiple modules?

What do you think would be better?

Reply:
BedrockError is defined once in bedrock lib and reused elsewhere.  Here it's two classes with the same name in two libs.  It might not be an issue as these are private (I think?) so no one would be instantiating the errors directly.  I was just thinking there could be issues if someone wanted to use `instanceof` to check for this error and would maybe have to check multiple similarly named classes.

An option would be to expose the error class from one lib and reuse it in the other.

Reply:
@dlongley what are your thoughts on this topic?

Reply:
@mattcollier,

One option would be to expose `HTTPSignatureError` in http-signature-header (which I believe is the "lowest level" of all three libs) and import it into http-signature-crypto and here. Then it would be the same error class.